packit_service/models.py

```diff
@@ -891,20 +911,23 @@ def get_all(cls) -> Optional[Iterable["InstallationModel"]]:
             return session.query(InstallationModel).all()
 
     @classmethod
-    def create(cls, event):
+    def create(cls, event: dict):
         with get_sa_session() as session:
-            installation = cls.get_by_account_login(event.account_login)
+            account_login = event.get("account_login")
+            installation = cls.get_by_account_login(account_login)
             if not installation:
                 installation = cls()
-                installation.account_login = event.account_login
-                installation.account_id = event.account_id
-                installation.account_url = event.account_url
-                installation.account_type = event.account_type
-                installation.sender_login = event.sender_login
-                installation.sender_id = event.sender_id
-                installation.created_at = event.created_at
+                installation.account_login = account_login
+                installation.account_id = event.get("account_id")
+                installation.account_url = event.get("account_url")
+                installation.account_type = event.get("account_type")
+                installation.sender_login = event.get("sender_login")
+                installation.sender_id = event.get("sender_id")
+                installation.created_at = datetime.fromtimestamp(
+                    event.get("created_at")
+                )
```

Comment:
Most of the attributes are not needed in the rest of InstallationHandler code, but yes,
we can:
1. parse all these attributes from dict in handler code to be able to use them here
2. create the db record when the InstallationEvent is parsed from webhook

Reply:
> create the db record when the InstallationEvent is parsed from webhook

Since this requires a database action, I would try to avoid any unnecessary action during parsing.

Reply:
changed